### PR TITLE
[depends] add Makefile target to download dav1d

### PR DESCRIPTION
**Pull request by wsnipex**
## Description

adds "make download" to the dav1d depends makefile, to allow downloading the source, same as with libdvd
Also allows local building without depends

## Motivation and Context

helps packaging

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [X] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [ ] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: tools/depends/target/dav1d/Makefile [hunk 1]
```diff
@@ -1,6 +1,24 @@
...
+-include ../../Makefile.include
 include DAV1D-VERSION
-DEPS = ../../Makefile.include Makefile DAV1D-VERSION
+DEPS = Makefile DAV1D-VERSION
+
+ifeq ($(CROSS_COMPILING), yes)
+  DEPS += ../../Makefile.include
+else
+  ROOT_DIR := $(shell dirname $(realpath $(lastword $(MAKEFILE_LIST))))
+  PREFIX := $(ROOT_DIR)/dvdcss-install
```
Comment (fuzzard) [inline tools/depends/target/dav1d/Makefile]: C&P error I guess?

File: tools/depends/target/dav1d/Makefile [hunk 2]
```diff
@@ -1,6 +1,24 @@
...
+
+  ifeq ($(PLATFORM),)
+    PLATFORM = native
+    TARBALLS_LOCATION = $(ROOT_DIR)
+  endif
+endif
+
+
+RETRIEVE_TOOL := curl -Ls --retry 10 --retry-delay 3 -o $(ARCHIVE)
+ARCHIVE_TOOL := tar --strip-components=1 -xf
```
Comment (lrusak) [inline tools/depends/target/dav1d/Makefile]: can you move these two inside the `else` block. They aren't needed when doing a depends build

**Review by lrusak — CHANGES_REQUESTED** [review]

File: tools/depends/target/dav1d/Makefile (hunk 2 above)
Comment (wsnipex) [reply tools/depends/target/dav1d/Makefile]: they are only set if not already defined.

File: tools/depends/target/dav1d/Makefile (hunk 1 above)
Comment (wsnipex) [reply tools/depends/target/dav1d/Makefile]: yep, thanks

**Comment by lrusak** [comment]
Do you mean to use `?=` then?

**Comment by wsnipex** [comment]
oh yes, indeed :)

**Comment by wsnipex** [comment]
updated

**Review by lrusak — APPROVED** [review]

**Pull request merged**